homeassistant/components/tuya/climate.py

```diff
@@ -0,0 +1,464 @@
+    "wind": CURRENT_HVAC_FAN,
+    "auto": CURRENT_HVAC_IDLE,
+}
+
+TUYA_SUPPORT_TYPE = {
+    "kt",  # Air conditioner
+    "qn",  # Heater
+    "wk",  # Thermostat
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up tuya climate dynamically through tuya discovery."""
+    _LOGGER.debug("climate init")
+
+    hass.data[DOMAIN][entry.entry_id][TUYA_HA_TUYA_MAP][
+        DEVICE_DOMAIN
+    ] = TUYA_SUPPORT_TYPE
+
+    @callback
+    def async_discover_device(dev_ids: list[str]) -> None:
+        """Discover and add a discovered tuya climate."""
+        _LOGGER.debug("climate add-> %s", dev_ids)
+        if not dev_ids:
+            return
+        entities = _setup_entities(hass, entry, dev_ids)
+        async_add_entities(entities)
+
+    entry.async_on_unload(
+        async_dispatcher_connect(
+            hass, TUYA_DISCOVERY_NEW.format(DEVICE_DOMAIN), async_discover_device
+        )
+    )
+
+    device_manager = hass.data[DOMAIN][entry.entry_id][TUYA_DEVICE_MANAGER]
+    device_ids = []
+    for (device_id, device) in device_manager.device_map.items():
+        if device.category in TUYA_SUPPORT_TYPE:
+            device_ids.append(device_id)
+    async_discover_device(device_ids)
+
+
+def _setup_entities(
+    hass: HomeAssistant, entry: ConfigEntry, device_ids: list[str]
+) -> list[Entity]:
+    """Set up Tuya Climate."""
+    device_manager = hass.data[DOMAIN][entry.entry_id][TUYA_DEVICE_MANAGER]
+    entities: list[Entity] = []
+    for device_id in device_ids:
+        device = device_manager.device_map[device_id]
+        if device is None:
+            continue
+        entities.append(TuyaHaClimate(device, device_manager))
+        hass.data[DOMAIN][entry.entry_id][TUYA_HA_DEVICES].add(device_id)
+    return entities
+
+
+class TuyaHaClimate(TuyaHaEntity, ClimateEntity):
+    """Tuya Switch Device."""
+
+    def __init__(
+        self, device: TuyaDevice, device_manager: TuyaDeviceManager
+    ) -> None:
+        """Init Tuya Ha Climate."""
+        super().__init__(device, device_manager)
+        if DPCODE_C_F in self.tuya_device.status:
+            self.dp_temp_unit = DPCODE_C_F
+        else:
+            self.dp_temp_unit = DPCODE_TEMP_UNIT_CONVERT
+
+    def get_temp_set_scale(self) -> int:
+        """Get temperature set scale."""
+        dp_temp_set = DPCODE_TEMP_SET if self.is_celsius() else DPCODE_TEMP_SET_F
+        temp_set_value_range = json.loads(
+            self.tuya_device.status_range.get(dp_temp_set).values
+        )
+        return temp_set_value_range.get("scale")
+
+    def get_temp_current_scale(self) -> int:
+        """Get temperature current scale."""
+        dp_temp_current = (
+            DPCODE_TEMP_CURRENT if self.is_celsius() else DPCODE_TEMP_CURRENT_F
+        )
+        temp_current_value_range = json.loads(
+            self.tuya_device.status_range.get(dp_temp_current).values
```

Comment:
There will be an AttributeError here if `dp_temp_current` is missing in the status range dict.